staging/src/k8s.io/cli-runtime/pkg/resource/builder.go

```diff
@@ -1186,6 +1194,17 @@ func HasNames(args []string) (bool, error) {
 	return hasCombinedTypes || len(args) > 1, nil
 }
 
+// expandIfFilePattern returns all the filenames that match the input pattern
+// or the filename if it is a specific filename and not a pattern.
+// If the input is a pattern and it yields no result it will result in an error.
+func expandIfFilePattern(pattern string) ([]string, error) {
+	matches, err := filepath.Glob(pattern)
```

Comment:
should we first call `filepath.Match(pattern, "")` to make sure this is a well-formed pattern Glob can handle and return `[]string{pattern}` for invalid patterns?

Reply:
Sure, instead I went with checking for `ErrBadPattern` since it is already baked in `Glob`